Code/Framework/AzCore/AzCore/Memory/IAllocator.h

```diff
@@ -28,17 +28,16 @@ namespace AZ
     class AllocatorManager;
 
     /**
-     * Allocator alloc/free basic interface. It is separate because it can be used
-     * for user provided allocators overrides
+     * Allocator schema interface
      */
-    class IAllocatorAllocate
+    class IAllocatorSchema
     {
     public:
         typedef void*           pointer_type;
         typedef size_t          size_type;
         typedef ptrdiff_t       difference_type;
 
-        virtual ~IAllocatorAllocate() {}
+        virtual ~IAllocatorSchema() {}
 
         virtual pointer_type            Allocate(size_type byteSize, size_type alignment, int flags = 0, const char* name = 0, const char* fileName = 0, int lineNum = 0, unsigned int suppressStackRecord = 0) = 0;
```

Comment:
Can `= default` the destructor and set `const char* name = nullptr, const char* fileName = nullptr`